djangocms_attributes_field/widgets.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from django.forms import Widget
+from django.forms.widgets import flatatt
+from django.utils.html import escape
+from django.utils.text import mark_safe
+
+
+class AttributesWidget(Widget):
+    """
+    A widget that displays key/value pairs from JSON as a list of text input
+    box pairs and back again.
+    """
+    # Heavily modified from a code snippet by Huy Nguyen:
+    # https://www.huyng.com/posts/django-custom-form-widget-for-dictionary-and-tuple-key-value-pairs
+    def __init__(self, *args, **kwargs):
+        """
+        Supports additional kwargs: `key_attr` and `val_attr`.
+        """
+        self.key_attrs = kwargs.pop('key_attrs', {})
+        self.val_attrs = kwargs.pop('val_attrs', {})
+        super(AttributesWidget, self).__init__(*args, **kwargs)
+
+    def _render_row(self, key, value, field_name, key_attrs, val_attrs):
+        """
+        Renders to HTML a single key/value pair row.
+
+        :param key: (str) key
+        :param value: (str) value
+        :param field_name: (str) String name of this field
+        :param key_attrs: (dict) HTML attributes to be applied to the key input
+        :param val_attrs: (dict) HTML attributes to be applied to the value input
+        """
+        return """
+        <div class="form-row attribute-pair">
+            <div class="field-box">
+               <label>Key</label>
```

Comment:
can we define a unique for="field-key-row-1" attribute associated with the input id="field-key-row-1" for each label/input pair and row?